### PR TITLE
[Sortable] Add support for attributes

### PR DESCRIPTION
**Pull request by vstelmakh**
Part of #2363

Apart of attributes support for Sortable, this PR contains typehints for `Gedmo\Tests\Tool\BaseTestCaseORM`, that's why there are so many files changed.

Sortable documentation is also updated/fixed.

### CODE REVIEW

File: tests/Gedmo/Tool/BaseTestCaseORM.php [hunk 1]
```diff
@@ -187,27 +173,23 @@ protected function stopQueryLog($dumpOnlySql = false, $writeToLog = false)
...
+    protected function getMetadataDriverImplementation(): MappingDriver
     {
         return new AnnotationDriver($_ENV['annotation_reader']);
     }
 
     /**
      * Get a list of used fixture classes
      *
-     * @return array
+     * @return array<class-string>
```
Comment (franmomu) [inline tests/Gedmo/Tool/BaseTestCaseORM.php:185]:
```suggestion
     * @phpstan-return list<class-string>
```

File: doc/sortable.md [hunk 2]
```diff
@@ -1,68 +1,55 @@
...
+|            | SortableGroup                             | SortablePosition                               |
+|------------|-------------------------------------------|------------------------------------------------|
+| Annotations | `@Gedmo\Mapping\Annotation\SortableGroup`   | `@Gedmo\Mapping\Annotation\SortablePosition`   |
+| Attributes | `#[Gedmo\Mapping\Annotation\SortableGroup]` | `#[Gedmo\Mapping\Annotation\SortablePosition]` |
+| Yaml       | `gedmo: [sortableGroup]`                    | `gedmo: [sortablePosition]`                    |
+| Xml        | `<gedmo:sortable-group />`                  | `<gedmo:sortable-position />`                  |
 
-- **@Gedmo\Mapping\Annotation\SortableGroup** it will use this field for **grouping**
-- **@Gedmo\Mapping\Annotation\SortablePosition** it will use this column to store **position** index
+> Implementin **[Sortable interface](../src/Sortable/Sortable.php) is optional**, except in cases there you need to identify entity as being Sortable.
```
Comment (franmomu) [inline doc/sortable.md:39]:
```suggestion
> Implementing **[Sortable interface](../src/Sortable/Sortable.php) is optional**, except in cases there you need to identify entity as being Sortable.
```

**Review by franmomu — COMMENTED** [review]
Thanks for the PR! 

We definitely need to update the general docs, this could be a way

File: tests/Gedmo/Tool/BaseTestCaseORM.php (hunk 1 above)
Comment (vstelmakh) [reply tests/Gedmo/Tool/BaseTestCaseORM.php:185]: Is it make sense to restrict this to phpstan specific doc types? Wouldn't be better to stick to generic ones?

Comment (franmomu) [reply tests/Gedmo/Tool/BaseTestCaseORM.php:185]: the thing is that `class-string` is not a generic one I think, looking that the usage in this library, it was always prefixed https://github.com/doctrine-extensions/DoctrineExtensions/search?q=class-string

Comment (vstelmakh) [reply tests/Gedmo/Tool/BaseTestCaseORM.php:185]: Make sense :+1:

File: doc/sortable.md [hunk 3]
```diff
@@ -1,68 +1,55 @@
...
-
-- [Including](#including-extension) the extension
-- Entity [example](#entity-mapping)
-- [Yaml](#yaml-mapping) mapping example
-- [Xml](#xml-mapping) mapping example
-- Basic usage [examples](#basic-examples)
-- Custom comparison [method](#custom-comparisons)
+- Annotation, Attribute, Yaml and Xml mapping support for extensions
 
+> Sortable, and all other doctrine extensions from this package, is available as Symfony bundle.
```
Comment (franmomu) [inline doc/sortable.md:11]:
```suggestion
> Sortable, and all other doctrine extensions from this package are available as Symfony bundle.
```

**Review by franmomu — APPROVED** [review]

File: doc/sortable.md [hunk 4]
```diff
@@ -1,68 +1,55 @@
...
+- [SortableGroup](../src/Mapping/Annotation/SortableGroup.php) - used to specify property for **grouping**
+- [SortablePosition](../src/Mapping/Annotation/SortablePosition.php) - used to specify property to store **position** index
 
-### Sortable annotations:
+|            | SortableGroup                             | SortablePosition                               |
+|------------|-------------------------------------------|------------------------------------------------|
+| Annotations | `@Gedmo\Mapping\Annotation\SortableGroup`   | `@Gedmo\Mapping\Annotation\SortablePosition`   |
+| Attributes | `#[Gedmo\Mapping\Annotation\SortableGroup]` | `#[Gedmo\Mapping\Annotation\SortablePosition]` |
+| Yaml       | `gedmo: [sortableGroup]`                    | `gedmo: [sortablePosition]`                    |
+| Xml        | `<gedmo:sortable-group />`                  | `<gedmo:sortable-position />`                  |
```
Comment (phansys) [inline doc/sortable.md:37]:
```suggestion
|             | SortableGroup                               | SortablePosition                               |
|-------------|---------------------------------------------|------------------------------------------------|
| Annotations | `@Gedmo\Mapping\Annotation\SortableGroup`   | `@Gedmo\Mapping\Annotation\SortablePosition`   |
| Attributes  | `#[Gedmo\Mapping\Annotation\SortableGroup]` | `#[Gedmo\Mapping\Annotation\SortablePosition]` |
| Yaml        | `gedmo: [sortableGroup]`                    | `gedmo: [sortablePosition]`                    |
| Xml         | `<gedmo:sortable-group />`                  | `<gedmo:sortable-position />`                  |
```

File: doc/sortable.md [hunk 5]
```diff
@@ -1,68 +1,55 @@
...
-**Note:** that you should register SortableRepository (or a subclass) as the repository in the Entity
+> You **should register [SortableRepository](../src/Sortable/Entity/Repository/SortableRepository.php)** (or a subclass) as the repository in the Entity
 annotation to benefit from its query methods.
 
-``` php
+### Annotation mapping example
+
+```php
 <?php
 namespace Entity;
```
Comment (phansys) [inline doc/sortable.md:49]:
```suggestion

namespace Entity;
```

**Review by phansys — CHANGES_REQUESTED** [review]

File: doc/sortable.md (hunk 5 above)
Comment (vstelmakh) [reply doc/sortable.md:50]: Resolved

**Review by phansys — APPROVED** [review]

**Comment by franmomu** [comment]
Thanks @vstelmakh!

**Pull request merged**